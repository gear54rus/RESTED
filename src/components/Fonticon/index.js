import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Fonticon({ icon, className }) {
  return (
    <i
      className={classNames('fas', `fa-${icon}`, className)}
      role="presentation"
    />
  );
}

Fonticon.propTypes = {
  icon: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default Fonticon;
