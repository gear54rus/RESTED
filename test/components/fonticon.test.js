import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

/* eslint-disable import/no-unresolved */
import Fonticon from 'components/Fonticon';

describe('Fonticon', () => {
  it('matches the previous snapshot', () => {
    const tree = renderer.create(
      <Fonticon icon="cog" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a font-awesome icon', () => {
    const tree = mount(
      <Fonticon icon="cog" />,
    );
    expect(tree.find('i').prop('className')).toBe('fas fa-cog');
  });

  it('has the role "presentation" for a11y', () => {
    const tree = mount(
      <Fonticon icon="cog" />,
    );
    expect(tree.find('i').prop('role')).toBe('presentation');
  });

  it('passes through className', () => {
    const tree = mount(
      <Fonticon icon="cog" className="test" />,
    );
    expect(tree.find('i').prop('className')).toBe('fas fa-cog test');
  });
});

