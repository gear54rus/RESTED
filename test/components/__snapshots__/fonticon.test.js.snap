// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fonticon matches the previous snapshot 1`] = `
<i
  className="fas fa-cog"
  role="presentation"
/>
`;
